src/lib/crypto/kimchi_backend/gadgets/test.ml

```diff
@@ -154,3 +155,211 @@ let%test_unit "custom gates integration" =
+    let open Foreign_field in
+    (* Initialize the SRS cache. *)
+    let () =
+      try Kimchi_pasta.Vesta_based_plonk.Keypair.set_urs_info [] with _ -> ()
+    in
+
+    (* Test the converter from Keccak format to ECDSA format.
+     * Tests the function bytes_to_standard_element
+     * together with standard_element_as_linear_combination_of_bytes
+     * and limb_as_linear_combination_of_bytes
+     * The circuit being created is the following:
+     * - 
+     *)
+    let test_converter ?cs bytestring endian =
+      let cs, _proof_keypair, _proof =
+        Runner.generate_and_verify_proof ?cs (fun () ->
+            let open Runner.Impl in
+            let secp256k1_modulus =
+              bignum_bigint_to_field_const_standard_limbs (module Runner.Impl)
+              @@ Common.bignum_bigint_of_hex
+                   "fffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f"
+            in
+
+            (* Set up snarky variables for input bytes *)
+            let bytestring =
+              List.map
+                ~f:(fun byte ->
+                  exists Field.typ ~compute:(fun () ->
+                      Field.Constant.of_int byte ) )
+                bytestring
+            in
+
+            let external_checks = External_checks.create (module Runner.Impl) in
+
+            let _elem =
+              Foreign_field.bytes_to_standard_element
+                (module Runner.Impl)
+                ~endian external_checks bytestring secp256k1_modulus 256
+            in
+
+            constrain_external_checks
+              (module Runner.Impl)
+              external_checks secp256k1_modulus ;
+            () )
+      in
+      cs
+    in
+
+    (* Test 1 limb *)
+    let _cs =
+      test_converter
+        [ 0x84; 0xc0; 0xd7; 0x28; 0xb6; 0x92; 0x8a; 0x0f; 0x4c; 0x29; 0xab ]
+        Keccak.Little
+    in
+    (* Test 2 limbs *)
+    let _cs =
+      test_converter
+        [ 0x28
+        ; 0x74
+        ; 0xab
+        ; 0x98
+        ; 0xc8
+        ; 0x23
+        ; 0xa8
+        ; 0xff
+        ; 0x4d
+        ; 0x91
+        ; 0xba
+        ; 0x81
+        ; 0x97
+        ; 0xc7
+        ; 0x81
+        ; 0xd6
+        ; 0x86
+        ; 0xa8
+        ; 0xd2
+        ; 0xb8
+        ; 0x27
+        ; 0x97
+        ]
+        Keccak.Little
+    in
+    (* Test 256 bits (f-1) *)
+    let cs =
+      test_converter
+        [ 0x2e
+        ; 0xfc
+        ; 0xff
+        ; 0xff
+        ; 0xfe
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ; 0xff
+        ]
+        Keccak.Little
+    in
+    print_endline "256 BITS > F" ;
```

Comment:
don't forget to remove the prints

Reply:
i had indeed forgotten as you can clearly tell